### PR TITLE
[CRI] Use DNSOptions passed by CRI in dockershim.

### PR DESCRIPTION
**Pull request by Random-Liu**
When @xlgao-zju is working on the CRI validation test, he found that dockershim is not using the DNSOptions passed in CRI. https://github.com/kubernetes-incubator/cri-tools/pull/30#issuecomment-290644357

This PR fixed the issue. I've manually tried, for `ClusterFirst` DNSPolicy, the resolv.conf will be:
```
nameserver 8.8.8.8
search corp.google.com prod.google.com prodz.google.com google.com
options ndots:5
```

For `Default` DNSPolicy, the resolv.conf will be:
```
nameserver 127.0.1.1
search corp.google.com prod.google.com prodz.google.com google.com
```

@xlgao-zju You should be able to test after this PR is merged.

/cc @yujuhong @feiskyer

### CODE REVIEW

File: pkg/kubelet/dockershim/docker_sandbox.go [hunk 1]
```diff
@@ -587,3 +589,53 @@ func toCheckpointProtocol(protocol runtimeapi.Protocol) Protocol {
 	glog.Warningf("Unknown protocol %q: defaulting to TCP", protocol)
 	return protocolTCP
 }
+
+// RewriteResolvFile rewrites resolv.conf file generated by docker.
```
Comment (yujuhong) [inline pkg/kubelet/dockershim/docker_sandbox.go]: nit: s/Rewrite/rewrite

File: pkg/kubelet/dockershim/docker_sandbox.go [hunk 2]
```diff
@@ -587,3 +589,53 @@ func toCheckpointProtocol(protocol runtimeapi.Protocol) Protocol {
...
+		glog.Errorf("ResolvConfPath is empty.")
+		return nil
+	}
+
+	if _, err := os.Stat(resolvFilePath); os.IsNotExist(err) {
+		return fmt.Errorf("ResolvConfPath %q does not exist", resolvFilePath)
+	}
+
+	var resolvFileContent []string
+
```
Comment (yujuhong) [inline pkg/kubelet/dockershim/docker_sandbox.go]: nit: remove the extra line

**Review by yujuhong — COMMENTED** [review]
LGTM with some nits.

**Comment by feiskyer** [comment]
LGTM. It should be much easier to clean up those codes after #43884.

**Comment by xlgao-zju** [comment]
@Random-Liu Yes, this commit will make cri-tools happy. :)

File: pkg/kubelet/dockershim/docker_sandbox.go (hunk 1 above)
Comment (Random-Liu) [inline pkg/kubelet/dockershim/docker_sandbox.go]: Done

File: pkg/kubelet/dockershim/docker_sandbox.go (hunk 2 above)
Comment (Random-Liu) [inline pkg/kubelet/dockershim/docker_sandbox.go]: Done

**Comment by Random-Liu** [comment]
@xlgao-zju FYI, this PR is merged.

**Pull request merged**